src/api/ledger/orders.js

```diff
@@ -26,10 +26,21 @@ function getOrdersAsync(account, options, callback) {
   validate.address(account);
   validate.getOrdersOptions(options);
 
-  const ledgerVersion = options.ledgerVersion
-                      || this.remote.getLedgerSequence();
+  if (!options.ledgerVersion) {
```

Comment:
I think it would simplify things a lot if we used the API's getLedgerVersion method and use promise.then. Also, could use getLedgerVersionHelper so that it only calls getLedgerVersion when ledgerVersion is not specified in the options. Then you can get rid of the branching here.